src/main/java/com/kobylynskyi/graphql/codegen/model/graphql/GraphQLRequestSerializer.java

```diff
@@ -37,8 +38,7 @@ public static String serialize(GraphQLRequest graphQLRequest) {
             builder.append(graphQLRequest.getResponseProjection().toString());
         }
         builder.append(" }");
-        builder.append("\"}");
-        return builder.toString();
+        return String.format("{\"query\":\"%s\"}", StringEscapeUtils.escapeJava(builder.toString()));
```

Comment:
What we technically want is to escape strings for JSON, not for Java, so we should respect the JSON specification instead, which is not the contract of `StringEscapeUtils.escapeJava()`.
That's why I was mentioning that maybe we should use a JSON library for this purpose (or ask the user to provide one).
As a temporary workaround though, this can work.

Reply:
@joffrey-bion Thank you so much for your input.
I've utilized Jackson library because we are already using it to parse external mapping config from son file (see `JsonMappingConfigSupplier`).